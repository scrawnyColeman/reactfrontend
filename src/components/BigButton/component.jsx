import React from 'react';
import styled from 'styled-components';
import { colours } from '../../constants/styles';
import { useHistory } from 'react-router-dom';

const StyledButton = styled.button`
    width: 100%;
    border: 6px solid ${colours.secondary}
    color: ${colours.secondary}
    background: none;
    border-radius: 25px;
    font-size: 2.5rem;
    padding: 1rem;

    &:hover {
        border: 6px solid ${colours.primary}
        background: none;
        color: ${colours.primary}
    }
`;
const BigButton = ({ text, path }) => {
    const history = useHistory();

    return <StyledButton onClick={() => history.push({ pathname: path })}>{text}</StyledButton>;
};
export default BigButton;
