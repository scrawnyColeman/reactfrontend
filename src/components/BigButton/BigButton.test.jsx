import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import BigButton from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('BigButton component', () => {
    const onClickMock = jest.fn();

    beforeEach(() => {
        onClickMock.mockReset();
    });

    test('BigButton matches snapshot', () => {
        const wrapper = shallow(<BigButton text="BigButton" path="path" />);
        expect(wrapper).toMatchSnapshot();
    });
});
