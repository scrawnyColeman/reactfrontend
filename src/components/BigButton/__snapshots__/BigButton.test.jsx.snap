// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigButton component BigButton matches snapshot 1`] = `
<styled.button
  onClick={[Function]}
>
  BigButton
</styled.button>
`;
